File: TechCorp_Vue.Server/Models/Employee.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace TechCorp_Vue.Server.Models;

[Table("Employee")]
public partial class Employee
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("name")]
    public string? Name { get; set; }

    [Column("positionid")]
    public int? Positionid { get; set; }

    [Column("managerid")]
    public int? Managerid { get; set; }

    [Column("idfile")]
    public int? Idfile { get; set; }

    [Column("departmentid")]
    public int? Departmentid { get; set; }

    [Column("createby")]
    public string? Createby { get; set; }

    [Column("createdate")]
    public DateTime? Createdate { get; set; }

    [Column("updateby")]
    public string? Updateby { get; set; }

    [Column("updatedate")]
    public DateTime? Updatedate { get; set; }

    [Column("isdeleted")]
    public bool? Isdeleted { get; set; }

    [Column("signatureid")]
    public int? Signatureid { get; set; }

    [ForeignKey("Departmentid")]
    [InverseProperty("Employees")]
    public virtual Department? Department { get; set; }

    [ForeignKey("Idfile")]
    [InverseProperty("Employees")]
    public virtual Filemodel? IdfileNavigation { get; set; }

    [InverseProperty("Manager")]
    public virtual ICollection<Employee> InverseManager { get; set; } = new List<Employee>();

    [ForeignKey("Managerid")]
    [InverseProperty("InverseManager")]
    public virtual Employee? Manager { get; set; }

    [ForeignKey("Positionid")]
    [InverseProperty("Employees")]
    public virtual Position? Position { get; set; }

    [ForeignKey("Signatureid")]
    [InverseProperty("Employees")]
    public virtual Signature? Signature { get; set; }
}

File: techcorp_vue.client/src/models/Employee.ts
import { reactive } from "vue";

export interface Department {
  id: number;
  name: string;

}

export interface Signature {
  id: number;
  signature: string;

}

export interface Position {
  id: number;
  title: string;
}

export interface Filemodel {
  id: number;
  filepath: string;
}

export interface Employee {
  id: number;
  name: string; // ต้องตรงกับ C# property
  CoverPhotoFile?: File;
  positionid?: number | null;
  managerid?: number | null;
  departmentid?: number | null;
  isdeleted?: boolean;
  idfile?: number | null;
  department?: Department | null;
  idfileNavigation?: Filemodel | null;
  position?: Position | null;
  inverseManager?: Employee[]; // ลูก


  signatureid?: number | null;
  signature?: { signature1: string }; 
}

export class EmployeeCreate {
  id = 0;
  name = "";
  managerid: number | null = null;
  departmentid: number | null = null;
  positionid: number | null = null;
  isdeleted = false;
  idfile: number | null = null;
  idfileNavigation?: Filemodel | null;
  CoverPhotoFile?: File;
  profileImage: string | null = null;
  signatureid: number | null = null;
  signature?: string;

  inverseManager: EmployeeCreate[] = [];
}
